Firestore/Source/Local/FSTLevelDBMigrations.mm

```diff
@@ -80,6 +99,16 @@ + (void)runMigrationsOnDB:(std::shared_ptr<DB>)db {
     case 0:
       EnsureTargetGlobal(db, group);
       // Fallthrough
+    case 1:
+      // We need to make sure we have metadata, since we're going to read and modify it
+      // in this migration. Commit the current transaction and start a new one. Since we're
+      // committing, we need to save a version. It's safe to save this one, if we crash
+      // after saving we'll resume from this step when we try to migrate.
+      SaveVersion(1, group);
```

Comment:
This is pretty gross and is another poke at the idea that we should make WriteGroups into Transactions that support reading back pending values.

(This is not actionable feedback here; more of a reminder for myself.)

Reply:
Noted. Not sure it's really worth doing yet, but I can see how the current state is less than optimal.